ports/renesas-ra/mphalport.h

```diff
@@ -35,6 +35,22 @@
 #define MICROPY_PY_PENDSV_ENTER   uint32_t atomic_state = raise_irq_pri(IRQ_PRI_PENDSV)
 #define MICROPY_PY_PENDSV_EXIT    restore_irq_pri(atomic_state)
 
+
+// Port level Wait-for-Event macro
+//
+// Do not use this macro directly, include py/runtime.h and
+// call mp_event_wait_indefinite() or mp_event_wait_ms(timeout)
+// Uses WFI when timeout is given so it wakes up from systick
+// otherwise it may not wake up within given timeout.
+#define MICROPY_INTERNAL_WFE(TIMEOUT_MS)                        \
+    do {                                                        \
+        if ((TIMEOUT_MS) < 0) {                                 \
+            __WFE();                                            \
```

Comment:
I'm not sure WFE is correct to use on this port.  It doesn't do SEV anywhere, and SCB_SCR_SEVONPEND is not enabled (as far as I can tell), so it's unlikely that the WFE will ever wake up.

Eg this WFE will be run each time `mp_hal_stdin_rx_chr()` loops around, and so it's supposed to wake whenever a new character is available, ie an "event" occurs.  But I can't see how that would work in the current code (would need to add SEV's in the right locations).

Reply:
Oh interesting, I don't know much about these internals.
I had done some minimal testing at repl (over uart) which I just repeated:
```
anl@STEP:~/micropython/ports/renesas-ra$ mpremote u0
Connected to MicroPython at /dev/ttyUSB0
Use Ctrl-] or Ctrl-x to exit this shell

>>>
MPY: sync filesystems
MPY: soft reboot
MicroPython v1.26.0-preview.25.g0f1c2c1f25.dirty on 2025-04-24; EK-RA4W1 with RA4W1
Type "help()" for more information.
>>>
>>> import time
>>> time.sleep(1)
>>> while True:
...   time.sleep(1)
...
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
KeyboardInterrupt:
>>>
>>> while True:
...   pass
...
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
KeyboardInterrupt:
>>>
```

Reply:
It may be that interrupts automatically set the event flag on these MCUs.  In that case, WFI and WFE are equivalent, in which case this macro should just unconditionally use one of them.

We need to get this right because it's tricky and can have lots of impacts on the system if not done the right way.

Reply:
I went hunting on their website for a reference manual that covers the core in this way, found nothing.

Looking at the source, these do just both go to cmsis that implements them as the two `wfi`/`wfe` assembly commands which doesn't really help. 

Based on the lack of data I'm certainly inclined to take the route of simple / safe and just use the one-liner WFI command instead.

Reply:
Yes, I think just a simple WFI would be best here.

Reply:
I've pushed the simplified change now thanks.